kura/org.eclipse.kura.linux.net/src/main/java/org/eclipse/kura/linux/net/dns/LinuxNamed.java

```diff
@@ -178,39 +159,34 @@ public boolean enable() throws KuraException {
         // write config happened during 'set config' step
         try {
             // Check if named is running
-            int pid = LinuxProcessUtil.getPid(procString);
+            int pid = LinuxProcessUtil.getPid(this.procString);
             if (pid > -1) {
                 // If so, disable it
                 logger.error("DNS server is already running, bringing it down...");
                 disable();
             }
             // Start named
             int result;
-            if (OS_VERSION.equals(
-                    KuraConstants.Mini_Gateway.getImageName() + "_" + KuraConstants.Mini_Gateway.getImageVersion())
-                    || OS_VERSION.equals(KuraConstants.ReliaGATE_10_05.getImageName() + "_"
-                            + KuraConstants.ReliaGATE_10_05.getImageVersion())
-                    || OS_VERSION.equals(KuraConstants.Intel_Edison.getImageName() + "_"
-                            + KuraConstants.Intel_Edison.getImageVersion() + "_"
-                            + KuraConstants.Intel_Edison.getTargetName())) {
+            if (OS_VERSION
+                    .startsWith(KuraSupportedPlatforms.YOCTO_121.getImageName() + "_")
+                    || OS_VERSION.startsWith(KuraSupportedPlatforms.YOCTO_161.getImageName() + "_"
+                            + KuraSupportedPlatforms.YOCTO_161.getImageVersion() + "_")) {
                 result = LinuxProcessUtil.start("/etc/init.d/bind start");
-            } else if (OS_VERSION.equals(KuraConstants.Raspberry_Pi.getImageName())
-                    || OS_VERSION.equals(KuraConstants.BeagleBone.getImageName())
-                    || OS_VERSION.equals(KuraConstants.ReliaGATE_50_21_Ubuntu.getImageName() + "_"
-                            + KuraConstants.ReliaGATE_50_21_Ubuntu.getImageVersion())) {
+            } else if (OS_VERSION.equals(KuraSupportedPlatforms.RASPBIAN_100.getImageName())
+                    || OS_VERSION.equals(KuraSupportedPlatforms.DEBIAN_100.getImageName())) {
                 result = LinuxProcessUtil.start("/etc/init.d/bind9 start");
-            } else if (OS_VERSION.equals(KuraConstants.Fedora_Pi.getImageName())
-                    || OS_VERSION.equals(KuraConstants.Intel_Up2.getImageName())
-                    || OS_VERSION.equals(KuraConstants.Reliagate_20_26.getImageName() + "_"
-                            + KuraConstants.Reliagate_20_26.getImageVersion())) {
+            } else if (OS_VERSION.equals(KuraSupportedPlatforms.FEDORA_2X.getImageName())
```

Comment:
Wouldn't it make sense to provide some kind of `NameServerController` interface and then implement that by some `SystemdNameServerController` … instead of keeping that if-nightmare alive?!

Reply:
That is the plan. This is just a first step. I believe I've talked about this in one of the last weekly meetings.